x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback } from 'react';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiTextArea,
+  EuiHorizontalRule,
+  EuiText,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+  EuiComboBox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import {
+  OperatingSystem,
+  ConditionEntryField,
+  isPathValid,
+  hasSimpleExecutableName,
+} from '@kbn/securitysolution-utils';
+import { isOneOfOperator } from '@kbn/securitysolution-list-utils';
+
+import { OS_TITLES } from '../../../../common/translations';
+import { ArtifactFormComponentProps } from '../../../../components/artifact_list_page';
+import {
+  CONDITIONS_HEADER,
+  CONDITIONS_HEADER_DESCRIPTION,
+  CONDITION_FIELD_DESCRIPTION,
+  CONDITION_FIELD_TITLE,
+  DESCRIPTION_LABEL,
+  DETAILS_HEADER,
+  DETAILS_HEADER_DESCRIPTION,
+  FIELD_LABEL,
+  NAME_LABEL,
+  OPERATOR_LABEL,
+  POLICY_SELECT_DESCRIPTION,
+  SELECT_OS_LABEL,
+  VALUE_LABEL,
+  ERRORS,
+} from '../../translations';
+import {
+  EffectedPolicySelect,
+  EffectedPolicySelection,
+} from '../../../../components/effected_policy_select';
+import {
+  GLOBAL_ARTIFACT_TAG,
+  BY_POLICY_ARTIFACT_TAG_PREFIX,
+} from '../../../../../../common/endpoint/service/artifacts/constants';
+import { useLicense } from '../../../../../common/hooks/use_license';
+import { isValidHash } from '../../../../../../common/endpoint/service/trusted_apps/validations';
+
+interface BlocklistEntry {
+  field: ConditionEntryField;
+  operator: 'included';
+  type: 'match_any';
+  value: string[];
+}
+
+interface ItemValidation {
+  name?: React.ReactNode[];
+  value?: React.ReactNode[];
+}
+
+function createValidationMessage(message: string) {
+  return <div>{message}</div>;
+}
+
+function getDropdownDisplay(field: ConditionEntryField) {
+  return (
+    <>
+      {CONDITION_FIELD_TITLE[field]}
+      <EuiText size="xs" color="subdued">
+        {CONDITION_FIELD_DESCRIPTION[field]}
+      </EuiText>
+    </>
+  );
+}
+
+export function CreateBlockListForm({
+  item,
+  policiesRequest,
+  onChange,
+}: ArtifactFormComponentProps) {
+  const [visited, setVisited] = useState({ name: false, value: false });
+  const [warnings, setWarnings] = useState({} as ItemValidation);
+  const [errors, setErrors] = useState({} as ItemValidation);
+
+  const isValid = useMemo(() => {
+    const errorCount = Object.values(errors)
+      .flat()
+      .reduce((memo, error) => memo + (error.length || 0), 0);
+    return errorCount === 0;
```

Comment:
So you just need to know if there is at least one error? is that right? if so, the above code seems to be a bit inefficient since it seem to loop through all errors.  You can just exit the moment you see one - maybe use `.some()`?